packages/core/src/common/performance/measurement-protocol.ts

```diff
@@ -0,0 +1,138 @@
+/********************************************************************************
+* Copyright (c) 2021 STMicroelectronics and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* This Source Code may also be made available under the following Secondary
+* Licenses when the conditions for such availability set forth in the Eclipse
+* Public License v. 2.0 are satisfied: GNU General Public License, version 2
+* with the GNU Classpath Exception which is available at
+* https://www.gnu.org/software/classpath/license.html.
+*
+* SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+*******************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { inject, injectable, LazyServiceIdentifer } from 'inversify';
+import { Measurement, MeasurementOptions } from './measurement';
+import { Stopwatch } from './stopwatch';
+import { JsonRpcServer } from '../messaging';
+
+export const BackendStopwatch = Symbol('BackendStopwatch');
+
+/** API path of the stopwatch service that exposes the back-end stopwatch to clients. */
+export const stopwatchPath = '/services/stopwatch';
+
+/** Token representing a remote measurement in the {@link BackendStopwatch} protocol. */
+export type RemoteMeasurement = number;
+
+export const BackendStopwatchOptions = Symbol('BackendStopwatchOptions');
+
+/**
+ * Injectable options for configuration of the (remote) back-end stopwatch service.
+ */
+export interface BackendStopwatchOptions {
+    /**
+     * Wether to disable the back-end stopwatch service. When `disabled`, the service interface
+     * is still fulfilled, but it just doesn't do anything. The default `disabled` state is
+     * `false` (being enabled).
+     */
+    disabled?: boolean;
+}
```

Comment:
I'm not sure I see the purpose of this interface. Without it, adopters who don't want a BackendStopwatch can do something like

```
rebind(BackendStopwatch).to(NullBackendStopwatch).inSingletonScope();
```
which seems at least as straightforward as
```
bind(BackendStopwatchOptions).toConstantValue({disabled: true});
```